src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetDateCommand.cs

```diff
@@ -212,26 +235,73 @@ public int Millisecond
         /// <summary>
         /// Unix format string.
         /// </summary>
-        [Parameter(ParameterSetName = "UFormat")]
+        [Parameter]
         [ValidateNotNullOrEmpty]
-        public string UFormat { get; set; }
+        public string UFormat
+        {
+            get
+            {
+                return _uFormat;
+            }
+
+            set
+            {
+                _uFormat = value;
+                _uFormatSpecified = true;
+            }
+        }
+
+        private string _uFormat;
+        private bool _uFormatSpecified;
 
         /// <summary>
-        /// Unix format string.
+        /// .NET format string.
         /// </summary>
-        [Parameter(ParameterSetName = "net")]
+        [Parameter]
         [ArgumentCompletions("FileDate", "FileDateUniversal", "FileDateTime", "FileDateTimeUniversal")]
-        public string Format { get; set; }
+        public string Format
+        {
+            get
+            {
+                return _format;
+            }
+
+            set
+            {
+                _format = value;
+                _formatSpecified = true;
+            }
+        }
+
+        private string _format;
+        private bool _formatSpecified;
 
         /// <summary>
         /// Gets or sets a value that converts date to UTC before formatting.
         /// </summary>
-        [Parameter(ParameterSetName = "net")]
+        [Parameter]
         public SwitchParameter AsUTC { get; set; }
         #endregion
 
         #region methods
 
+        /// <summary>
+        /// Validate arguments.
+        /// </summary>
+        protected override void BeginProcessing()
+        {
+            if (_formatSpecified && _uFormatSpecified)
```

Comment:
I think we should use parameter sets instead so as to let the parameter binder enforce mutually exclusive parameter; while their number can get out of hand, here we only need 4: `DateTime` (with  optional `-Format`), `DateTimeUFormat`, `UnixTime`  (with  optional `-Format`), and `UnixTimeUFormat`